py-polars/polars/lazyframe/frame.py

```diff
@@ -2721,6 +2722,26 @@ def sink_csv(
                 at any point without it being considered a breaking change.
         retries
             Number of retries if accessing a cloud instance fails.
+        engine
+            Select the engine used to write the query result, optional.
+            If set to `"cpu"` (default), the query result is written using the
```

Comment:
Our "cpu" engine typically means our in-memory engine, this doesn't support sinking without collecting into memory first. I think this should only have "streaming" and "gpu" options.